index.js

```diff
@@ -58,21 +59,22 @@ async function fastifyRateLimit (fastify, settings) {
   }, settings.addHeadersOnExceeding)
 
   // Global maximum allowed requests
-  globalParams.max = ((typeof settings.max === 'number' && !isNaN(settings.max)) || typeof settings.max === 'function')
+  globalParams.max = ((typeof settings.max === 'number' && !Number.isNaN(settings.max) && (settings.max = Math.trunc(settings.max)) >= 0) || typeof settings.max === 'function')
     ? settings.max
     : defaultMax
 
   // Global time window
   globalParams.timeWindow = typeof settings.timeWindow === 'string'
     ? ms.parse(settings.timeWindow)
-    : typeof settings.timeWindow === 'number' && !isNaN(settings.timeWindow)
-      ? settings.timeWindow
-      : 1000 * 60
-  globalParams.timeWindowInSeconds = Math.floor(globalParams.timeWindow / 1000)
+    : typeof settings.timeWindow === 'number' && !Number.isNaN(settings.timeWindow) && settings.timeWindow >= 0
```

Comment:
set timeWindow to Infinity. kaboom?

Reply:
Nice, everyone gets a certain number of requests and then they're banned for life 🤣

You think that scenario is absurd?

Reply:
Actually I don't even know if it would work in redis or not so I'm not going to risk it